scipy/stats/_continuous_distns.py

```diff
@@ -3079,6 +3079,90 @@ def _stats(self, mu):
 invgauss = invgauss_gen(a=0.0, name='invgauss')
 
 
+class norminvgauss_gen(rv_continuous):
+    r"""A Normal Inverse Gaussian continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `norminvgauss` is:
+
+    .. math::
+
+        f(x; a, b) = 
+        \frac{a  \exp\left(\sqrt{a^2 - b^2} + b x\right)}
+        {\pi \sqrt{1 + x^2}} \, K_1\left(a * sqrt{1 + x^2}\right)
+
+    where `x` is a real number, the parameter `a` is the tail heaviness
+    and `b` is the asymmetry parameter satisfying `a > 0` and `abs(b) <= a`.
+    `K_1` is the modified Bessel function of second kind (`scipy.special.k1`).
+
+    Due to the scaling properties of the distributions, note that
+
+    .. math::
+
+        norminvgauss.pdf(x, a d, b d, loc=\mu, scale=d) =
+        \frac{a  d \exp\left(d \sqrt{a^2 - b^2} + b (x - \mu)\right)}
+        {\pi \sqrt{d^2 + (x - \mu)^2}} \,
+        K_1\left(a sqrt{d^2 + (x - \mu)^2}\right),
+
+    which is another frequently used parametrisation of the distribution using
+    an additional scale and location parameter.
+
+    %(after_notes)s
+
+    A normal inverse Gaussian random variable with parameters `a` and `b` can
+    be expressed  as `X = b * V + sqrt(V) * X` where `X` is `norm(0,1)`
+    and `V` is `invgauss(mu=1/sqrt(a**2 - b**2))`.
+
+    When the parameter `mu` of `invgauss` is too small (`mu <= 0.0028`),
+    NaNs are returned. When sampling from the normal inverse Gaussian, this
+    is the case if `a**2 - b**2 >= 1/0.0028**2`.
```

Comment:
I find this paragraph confusing. Is this a limitation of the current implementation or something fundamental? Can this be worked around or is it not worth it in your opinion? What is the status of this magic number `0.0028`?

Reply:
It looks like the limitation of `numpy.random.wald` is no more there: https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/distributions.c

So I think I can remove it